### PR TITLE
isQuorumSetSane: Add error string parameter

### LINKED ISSUE #2723: Making isQuorumSetSane more informative
Hi,

We have some downstream changes to SCP which we'd like to upstream into `stellar-core`. One of them is making `isQuorumSetSane` more informative so it's possible to see why a quorum set is considered unsafe to use.

In our fork we used `char**` for this reason (https://github.com/bpfkorea/agora/pull/549), however I can see alternative implementions:

- Use `std::string` for the `reason` parameter
- Alternatively, use logging instead

Do you have any preference, and would you accept such a feature if we provided the patch?

Thanks!

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Please feel free to suggest improvements. Thanks!

Resolves #2723

### CODE REVIEW

File: src/scp/BallotProtocol.cpp
Comment (AndrejMitrovic) [inline src/scp/BallotProtocol.cpp:254]: Can someone suggest a way to inline this into the previous `CLOG` call?

File: src/main/Config.cpp
Comment (MonsieurNicolas) [inline src/main/Config.cpp:1265]: why not just keep `fmt::format("Invalid QUORUM_SET: {}", errString)`

File: src/scp/BallotProtocol.cpp
Comment (MonsieurNicolas) [reply src/scp/BallotProtocol.cpp:254]: I would just write this as
`CLOG(DEBUG, "SCP") << "Invalid quorum set received : " << (errString?errString:"<empty>");` and be done with it (also avoids a new variable).

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]
thanks. just a couple suggestions

**Comment by MonsieurNicolas** [comment]
Also, before we can merge your changes, please fill in the agreement per https://github.com/stellar/.github/blob/master/CONTRIBUTING.md#submitting-changes

**Comment by AndrejMitrovic** [comment]
Thanks! I've filled in the agreement and applied the changes.

**Comment by MonsieurNicolas** [comment]
Thank you!

**Comment by MonsieurNicolas** [comment]
r+ 8e2a98896b95de94acf99a6ab92fe9e878b08695

**Comment by AndrejMitrovic** [comment]
I've re-run clang formatter on the code. Sorry about that!

**Comment by MonsieurNicolas** [comment]
r+ e44e88a11c22db126481d303911781db70bdd7d0

**Comment by AndrejMitrovic** [comment]
It looks like Travis uses clag-format 5, and I used the one installed with brew (v10) which has a slightly different output. I'll try to get v5 installed and re-run the formatter.

**Comment by AndrejMitrovic** [comment]
Installing clang-format 5 seems to be tediously slow (tried https://gist.github.com/ffeu/6ffb75d8e8c7d92c0fbeb4b036599c33), so I applied to diff manually based on the results from Travis.

**Comment by Geod24** [comment]
@MonsieurNicolas : Since you got Travis set up, I wonder, why doesn't it trigger on PRs? Not having the CI trigger before approval (and only once per committer's approval) can make contributing quite tedious.

**Comment by MonsieurNicolas** [comment]
@AndrejMitrovic yes this is the way to do it, the error message is an actual patch that can be applied if you don't have clang-format-5 installed. Let's see if it works this time around...

**Comment by MonsieurNicolas** [comment]
@Geod24 good idea. I just enabled it for future PRs, from what I remember it was just too slow (we had it enabled a long time ago), but maybe it's not that bad now.

**Comment by MonsieurNicolas** [comment]
r+ cbefaa9b1cddc2abcb765be44f4a6d6ed9724d46

**Pull request merged**
